### DISCUSSION TITLE
Why not remove `init` and `Flags` and have `Application::run` take a user-constructed model directly?

### DISCUSSION THREAD

**Discussion by jjant** [hecrj/iced#1325] [Q&A] [answered]
Basically the title, in elm this makes sense because the flags are "provided" to you behind the scenes, but here you're responsible for setting up everything (i.e, writing `main`), wouldn't it simplify things to have `run` take an initial value (which users can create with a regular `MyApp::new(...)` pattern) and remove the Flags associated type?

**Comment by hecrj** [comment, marked as answer]
What about the `Command` produced by `Application::new`? We could ask for it in `run`, but I believe it would make the architecture a bit inconsistent.

The current design also makes it clearer that you are giving up complete control of the application lifecycle to the `iced` runtime, since you do not get a chance to have an application instance at any point in `main`. You do not even get to pick when the application is actually constructed.

**Reply by hecrj** [reply]
Furthermore, the `Application` trait fully implements The Elm Architecture which should help in the future if we ever want to allow composability at the application level.
